frontend/src/pages/home/index.tsx

```diff
@@ -3,25 +3,63 @@ import '@style/global.css';
 import '@components/style.css';
 
 import { Col, Row } from 'react-bootstrap';
+import { useQuery } from '@tanstack/react-query';
 
 import News from '@components/News';
 import TButton from '@components/TButton';
 import GoodsItem from '@components/GoodsItem';
 
-import newsData from '@pages/home/newsData.json';
-import goodsData from '@pages/discover/goodsData.json';
 import { useAuth } from '@lib/Auth';
 
 import TitleImgUrl from '@assets/images/title.png';
-import NewsImgUrl1 from '@assets/images/news1.jpg';
-import NewsImgUrl2 from '@assets/images/news2.jpg';
-import NewsImgUrl3 from '@assets/images/news3.jpg';
+import { CheckFetchStatus } from '@lib/Status';
+import { NewsProps } from '@components/News';
+import { GoodsItemProps } from '@components/GoodsItem';
 
 const Home = () => {
   const token = useAuth();
 
-  // TODO: remove this when integrating with backend
+  const { status: newsStatus, data: newsData } = useQuery({
+    queryKey: ['news'],
+    queryFn: async () => {
+      const response = await fetch(`/api/news`, {
+        headers: {
+          Accept: 'application/json',
+        },
+      });
+      if (!response.ok) {
+        throw new Error('Network response was not ok in news');
```

Comment:
try using ``RouteOnNotOk`` from ``@lib/Status``
uhh the un-merged version can choose to alart instead of forced route to error pages
this is a suggestion \:P